velox/exec/Driver.cpp

```diff
@@ -788,6 +788,8 @@ void Driver::run(std::shared_ptr<Driver> self) {
 
     case StopReason::kYield:
       // Go to the end of the queue.
+      VELOX_CHECK(self->state().hasBlockingFuture);
+      self->state().hasBlockingFuture = false;
```

Comment:
I wonder if that's the right fix. Should we instead avoid creating BlockingState in the first place?

```
        if (FOLLY_UNLIKELY(shouldYield())) {
          recordYieldCount();
          stop = StopReason::kYield;
          future = ContinueFuture{folly::Unit{}};
          blockingState = std::make_shared<BlockingState>(
              self, std::move(future), op, blockingReason_);
          guard.notThrown();
          return stop;
        }
```

Reply:
@mbasmanova you are right. We don't need it